src/lib/operators/jit_operator/jit_operations.hpp

```diff
@@ -0,0 +1,195 @@
+    catching_func(lhs.as<JIT_GET_DATA_TYPE(0, types)>(), rhs.as<JIT_GET_DATA_TYPE(1, types)>(), result);             \
+    break;
+
+#define JIT_COMPUTE_TYPE_CASE(r, types)                                                                              \
+  case static_cast<uint8_t>(JIT_GET_ENUM_VALUE(0, types)) << 8 | static_cast<uint8_t>(JIT_GET_ENUM_VALUE(1, types)): \
+    return catching_func(JIT_GET_DATA_TYPE(0, types)(), JIT_GET_DATA_TYPE(1, types)());
+
+/* Arithmetic operators */
+const auto jit_addition = [](const auto& a, const auto& b) -> decltype(a + b) { return a + b; };
+const auto jit_subtraction = [](const auto& a, const auto& b) -> decltype(a - b) { return a - b; };
+const auto jit_multiplication = [](const auto& a, const auto& b) -> decltype(a * b) { return a * b; };
+const auto jit_division = [](const auto& a, const auto& b) -> decltype(a / b) { return a / b; };
+const auto jit_modulo = [](const auto& a, const auto& b) -> decltype(a % b) { return a % b; };
+const auto jit_power = [](const auto& a, const auto& b) -> decltype(std::pow(a, b)) { return std::pow(a, b); };
+
+/* Comparison operators */
+const auto jit_equals = [](const auto& a, const auto& b) -> decltype(a == b, uint8_t()) { return a == b; };
+const auto jit_not_equals = [](const auto& a, const auto& b) -> decltype(a != b, uint8_t()) { return a != b; };
+const auto jit_less_than = [](const auto& a, const auto& b) -> decltype(a < b, uint8_t()) { return a < b; };
+const auto jit_less_than_equals = [](const auto& a, const auto& b) -> decltype(a <= b, uint8_t()) { return a <= b; };
+const auto jit_greater_than = [](const auto& a, const auto& b) -> decltype(a > b, uint8_t()) { return a > b; };
+const auto jit_greater_than_equals = [](const auto& a, const auto& b) -> decltype(a >= b, uint8_t()) { return a >= b; };
+
+const auto jit_like = [](const std::string& a, const std::string& b) -> uint8_t {
+  const auto regex_string = LikeTableScanImpl::sqllike_to_regex(b);
+  const auto regex = std::regex{regex_string, std::regex_constants::icase};
+  return std::regex_match(a, regex);
+};
+
+const auto jit_not_like = [](const std::string& a, const std::string& b) -> uint8_t {
+  const auto regex_string = LikeTableScanImpl::sqllike_to_regex(b);
+  const auto regex = std::regex{regex_string, std::regex_constants::icase};
+  return !std::regex_match(a, regex);
+};
+
+// The InvalidTypeCatcher acts as a fallback inplementation, if template specialization
+// fails for a type combination.
+template <typename F, typename R>
+struct InvalidTypeCatcher : F {
+  explicit InvalidTypeCatcher(F f) : F(std::move(f)) {}
+
+  using F::operator();
+
+  template <typename... Ts>
+  R operator()(const Ts...) const {
+    Fail("invalid combination of types for operation");
+  }
+};
+
+template <typename T>
+__attribute__((noinline)) void jit_compute(const T& op_func, const JitMaterializedValue& lhs,
```

Comment:
Why noinline?

Reply:
This is jit engine related. The compute functions get quite big due to the switch statement. When the jit inlines it itself during code specialization, it can prune all but the relevant case. This allows for faster jitting. I will add a comment to explain that.